src/spikeinterface/preprocessing/deepinterpolation/generators.py

```diff
@@ -41,16 +34,16 @@ def __init__(
                 r.get_num_channels() == recordings[0].get_num_channels() for r in recordings[1:]
             ), "All recordings must have the same number of channels"
 
-        total_num_samples = np.sum(r.get_total_samples() for r in recordings)
+        total_num_samples = np.sum([r.get_total_samples() for r in recordings])
```

Comment:
```suggestion
        total_num_samples = sum([r.get_total_samples() for r in recordings])
```

Sorry, too much talking and too few suggestions : P

I think this is the easiest!